src/core/toga/widgets/optioncontainer.py

```diff
@@ -137,10 +141,26 @@ def content(self):
         """
         return self._content
 
+    @property
+    def number_of_tabs(self):
+        return len(self.content)
+
+    @property
+    def current_tab_index(self):
+        return self._impl.get_current_tab_index()
+
+    @current_tab_index.setter
+    def current_tab_index(self, current_tab_index):
+        self._impl.set_current_tab_index(current_tab_index)
+
+    @property
+    def current_tab(self):
+        return self.content[self.current_tab_index]
```

Comment:
This is the piece that is grating against me; details in the top level review comment.